### DISCUSSION TITLE
Passing a string to tar_read?

### DISCUSSION THREAD

**Discussion by mike-lawrence** [ropensci/targets#354] [Help] [answered]
I'm probably misusing things somewhere, but I'm playing with the excellent stantargets with a [static branching pattern to explore multiple models under different data size conditions](https://github.com/wlandau/stantargets/discussions/29#discussioncomment-509439), and I can get the list of resulting targets by filtering the metadata, but I'm having trouble then programatically retrieving the corresponding objects using `tar_read` presumably thanks to tidy-eval stuff I'm still not great with. Here's what I have, with things I've tried as commented-out entries at the end:

```r
(
  tar_meta()
  %>% filter(
    type=='pattern'
    , !str_contains(name,'_data')
  ) 
  %>% pull(name)
  #works up to here

  # %>% map_dfr( .f = tar_read ) #doesn't work

  # %>% map_dfr( .f = tar_read(!!.) ) #doesn't work

  # %>% map_dfr( .f = function(x) { tar_read(!!x) } ) #doesn't work  
)
```

**Comment by wlandau** [comment, marked as answer]
`tar_read_raw()` can accept a string. Alternatively, `tar_load()` accepts `tidyselect` syntax. To create a list using `tar_load()`:

```r
envir <- new.env()
tar_load(ends_with("_data"), envir = envir)
as.list(envir)
```
